test/types/string/dmk/oobbyte2.prediff

```diff
@@ -0,0 +1,4 @@
+#! /bin/bash
+
+sed -e 's/^.*String\.chpl:[0-9]*: //' <$2 >$2.tmp
```

Comment:
It seems that these error line numbers should be in the user's code. I believe you can achieve that with `pragma "insert line file info" pragma "always propagate line file info"` on the function in question but this case might need some other handling to do with the primitive error handling. But couldn't we make the compiler code with the primitive raise only internal errors and make the user-facing `compilerError`s occur in the modules code implementing `ascii` and `toByte` ? It looks like the `-1` case which is handled that way is giving the right line number for the error...

Reply:
Unfortunately, the user errors must remain within the primitive because we don't know a string's length in param Chapel code.

I tried the pragmas, but they don't seem to do anything in this case.

Reply:
> we don't know a string's length in param Chapel code.

I don't think that's right.

primitive.cpp shows `"string_length"` next to `"ascii"`, and in fact it is implemented in postFold just above ascii.

And this program works for me:

``` chapel
param len = "aa".length;

var tup:len*int;
writeln(len);
writeln(tup);
```

Reply:
Oh, that's very strange.  I can't use `string.len` in param code, but it does accept `string.length` which is a proc that references `string.len`.  I will see what I can do with this.

Reply:
I see what's going on.  There is a separate param `string.length` that calls the primitive.

I was trying to avoid `string.length` in case its meaning changes to count codepoints, which would break this code.

Reply:
You could call the primitive directly. It's likely that we'll need different primitives for the codepoint vs bytes cases, once we get the compiler working with those.

Reply:
OK, fixed.  Thanks.  I will retest today and merge this weekend.